### PR TITLE
Refactor Aggregator to avoid too many templates.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/DoubleSumAggregator.java [hunk 1]
```diff
@@ -18,21 +18,26 @@
...
 
   DoubleSumAggregator() {
-    this.value = new AtomicDouble();
+    this.current = new AtomicDouble(0.0);
   }
 
   @Override
   public void merge(DoubleSumAggregator other) {
-    this.value.addAndGet(other.value.get());
+    current.getAndAdd(other.current.get());
```
Comment (thisthat) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/DoubleSumAggregator.java]: I would prefer to have `this.current` for readability.
```suggestion
    this.current.getAndAdd(other.current.get());
```

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/LongSumAggregator.java [hunk 2]
```diff
@@ -18,21 +18,26 @@
...
 
   LongSumAggregator() {
-    this.value = new AtomicLong();
+    this.current = new AtomicLong(0L);
   }
 
   @Override
   public void merge(LongSumAggregator other) {
-    this.value.addAndGet(other.value.get());
+    current.getAndAdd(other.current.get());
```
Comment (thisthat) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/LongSumAggregator.java]: The same comment applies here

File: sdk/src/test/java/io/opentelemetry/sdk/metrics/LongSumAggregatorTest.java [hunk 3]
```diff
@@ -26,6 +26,6 @@
   @Test
   public void longSumAggregation() {
     LongSumAggregator longSumAggregator = new LongSumAggregator();
-    longSumAggregator.update(12);
+    longSumAggregator.recordLong(12);
   }
 }
```
Comment (thisthat) [inline sdk/src/test/java/io/opentelemetry/sdk/metrics/LongSumAggregatorTest.java]: Could you please add also a test class for `DoubleSumAggregator`?

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/DoubleSumAggregator.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/DoubleSumAggregator.java]: done.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/LongSumAggregator.java (hunk 2 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/LongSumAggregator.java]: done.

File: sdk/src/test/java/io/opentelemetry/sdk/metrics/LongSumAggregatorTest.java (hunk 3 above)
Comment (bogdandrutu) [reply sdk/src/test/java/io/opentelemetry/sdk/metrics/LongSumAggregatorTest.java]: Added, so far this is a trivial test until I defined the entire Aggregation class.

**Comment by bogdandrutu** [comment]
@jkwatson please review :)

**Comment by bogdandrutu** [comment]
@thisthat there are two reasons:
1) avoid double -> long conversion on the critical path.
2) long gives better precision for integer large numbers.

**Pull request merged**